src/python/turicreate/toolkits/activity_classifier/_tf_model_architecture.py

```diff
@@ -83,9 +89,9 @@ def __init__(self, net_params, batch_size, num_features, num_classes, prediction
             beta_initializer=_tf.initializers.constant(net_params['bn_beta'], verify_shape=True),
             gamma_initializer=_tf.initializers.constant(net_params['bn_gamma'], verify_shape=True),
             moving_mean_initializer=_tf.initializers.constant(net_params['bn_running_mean'], verify_shape=True),
-            moving_variance_initializer=_tf.initializers.constant(net_params['bn_running_var'], verify_shape=True), training=self.is_training )
+            moving_variance_initializer=_tf.initializers.constant(net_params['bn_running_var'], verify_shape=True), training=self.is_training)
         dense = _tf.nn.relu(dense)
-        dense = _tf.layers.dropout(dense, rate=0.5, training=self.is_training)
+        dense = _tf.layers.dropout(dense, rate=0.5, training=self.is_training, seed=seed)
```

Comment:
hmmm just found something strange and doesn't make sense
because we re-run a session every single time for each mini batch,
set a fixed seed either from graph-level or opt-level with produce the same result
every single time.
Say:
```
random_num = tf.random.unifrom([],0,1,seed=0)
with tf.Session() as session:
   print(session.run(random_num)

with tf.Session() as session:
   print(session.run(random_num))
```
will produce two identical random number.
That is to say, the behavior across all mini-batch will be the same.
@nickjong  @shreyajain17  I don't think this is what we expect.

Reply:
I don't really understand completely. But you get different results with different seeds and same result with the same seed. Isn't that what is expected?

```In [4]: model = tc.activity_classifier.create(data,'exp_id','activity',random_seed=1)
Pre-processing 672978 samples...
Using sequences of size 2000 for model creation.
Processed a total of 55 sessions.
Using GPU to create model (AMD Radeon Pro 560X)
+--------------+----------------+---------------------+--------------+
| Iteration    | Train Accuracy | Validation Accuracy | Elapsed Time |
+--------------+----------------+---------------------+--------------+
| 1            | 0.53856        | 0.767315            | 12.20s       |
| 2            | 0.745894       | 0.853582            | 13.84s       |
| 3            | 0.814608       | 0.892493            | 15.52s       |
| 4            | 0.85464        | 0.905128            | 17.16s       |
| 5            | 0.877809       | 0.910418            | 18.77s       |
| 6            | 0.885565       | 0.915482            | 20.52s       |
| 10           | 0.90907        | 0.918054            | 27.72s       |
+--------------+----------------+---------------------+--------------+
Training complete
Total Time Spent: 50.2019

In [5]: model = tc.activity_classifier.create(data,'exp_id','activity',random_seed=2)
Pre-processing 671511 samples...
Using sequences of size 2000 for model creation.
Processed a total of 55 sessions.
Using GPU to create model (AMD Radeon Pro 560X)
+--------------+----------------+---------------------+--------------+
| Iteration    | Train Accuracy | Validation Accuracy | Elapsed Time |
+--------------+----------------+---------------------+--------------+
| 1            | 0.546871       | 0.786748            | 12.26s       |
| 2            | 0.757693       | 0.855413            | 13.91s       |
| 3            | 0.829075       | 0.871994            | 15.56s       |
| 4            | 0.859569       | 0.894701            | 17.20s       |
| 5            | 0.881342       | 0.89708             | 18.88s       |
| 6            | 0.886383       | 0.905612            | 20.76s       |
| 10           | 0.906395       | 0.919267            | 27.95s       |
+--------------+----------------+---------------------+--------------+
Training complete
Total Time Spent: 50.7373

In [6]: model = tc.activity_classifier.create(data,'exp_id','activity',random_seed=1)
Pre-processing 672978 samples...
Using sequences of size 2000 for model creation.
Processed a total of 55 sessions.
Using GPU to create model (AMD Radeon Pro 560X)
+--------------+----------------+---------------------+--------------+
| Iteration    | Train Accuracy | Validation Accuracy | Elapsed Time |
+--------------+----------------+---------------------+--------------+
| 1            | 0.53856        | 0.767315            | 12.53s       |
| 2            | 0.745894       | 0.853582            | 14.22s       |
| 3            | 0.814608       | 0.892493            | 15.93s       |
| 4            | 0.85464        | 0.905128            | 17.62s       |
| 5            | 0.877809       | 0.910418            | 19.32s       |
| 6            | 0.885565       | 0.915482            | 21.13s       |
| 10           | 0.90907        | 0.918054            | 28.65s       |
+--------------+----------------+---------------------+--------------+```

Reply:
What I'm saying is,
say the input feature of the dropout layer has 10 dimensions, 
`dropout(seed=0)` with dropout the `dim=0`,
so for every mini batch the first dim will be dropped.
But the model can learn better if it change.

say we have '3' total mini batch
==now==
minibatch_1:drop dim=0
minibatch_2:drop dim=0
minibatch_10:drop dim=0

I'm just wondering we expect thing more like
minibatch_1:drop dim=0
minibatch_2:drop dim=3
minibatch_10:drop dim=4

and the dim [0,3,4] is fixed for every experiment if we provide the same seed.

Hope this clarify.

Reply:
So this is what tensorflow says - 
seeds can be set at graph-level and op-level.

To generate different sequences across sessions, set neither graph-level nor op-level seeds:
```
a = tf.random.uniform([1])
b = tf.random.normal([1])

print("Session 1")
with tf.compat.v1.Session() as sess1:
  print(sess1.run(a))  # generates 'A1'
  print(sess1.run(a))  # generates 'A2'
  print(sess1.run(b))  # generates 'B1'
  print(sess1.run(b))  # generates 'B2'

print("Session 2")
with tf.compat.v1.Session() as sess2:
  print(sess2.run(a))  # generates 'A3'
  print(sess2.run(a))  # generates 'A4'
  print(sess2.run(b))  # generates 'B3'
  print(sess2.run(b))  # generates 'B4'
```
To generate the same repeatable sequence for an op across sessions, set the seed for the op:
```
a = tf.random.uniform([1], seed=1)
b = tf.random.normal([1])

# Repeatedly running this block with the same graph will generate the same
# sequence of values for 'a', but different sequences of values for 'b'.
print("Session 1")
with tf.compat.v1.Session() as sess1:
  print(sess1.run(a))  # generates 'A1'
  print(sess1.run(a))  # generates 'A2'
  print(sess1.run(b))  # generates 'B1'
  print(sess1.run(b))  # generates 'B2'

print("Session 2")
with tf.compat.v1.Session() as sess2:
  print(sess2.run(a))  # generates 'A1'
  print(sess2.run(a))  # generates 'A2'
  print(sess2.run(b))  # generates 'B3'
  print(sess2.run(b))  # generates 'B4'
```

So wouldn't it be generating same sequence of dimensions rather than the same dimension  across sessions?

Reply:
You are right :).
We are using the same session to run multiple times haha.